doc/user-guide/data-structures.rst

```diff
@@ -495,6 +497,314 @@ dimension and non-dimension variables:
+    # Add the child Datatree to the root node
+    dt.children = {"child-node": dt2}
+    dt
+
+
+More idiomatically you can create a tree from a dictionary of ``Datasets`` and
+`DataTrees`. In this case we add a new node under ``dt["child-node"]`` by
+providing the explicit path under ``"child-node"`` as the dictionary key:
+
+.. ipython:: python
+
+    # create a third Dataset
+    ds3 = xr.Dataset({"zed": np.nan})
+    # create a tree from a dictionary of DataTrees and Datasets
+    dt = xr.DataTree.from_dict({"/": dt, "/child-node/new-zed-node": ds3})
+
+We have created a tree with three nodes in it:
+
+.. ipython:: python
+
+    dt
+
+
+
+Consistency checks are enforced. For instance, if we try to create a `cycle`,
+where the root node is also a child of a decendent, the constructor will raise
+an (:py:class:`~xarray.InvalidTreeError`):
+
+.. ipython:: python
+    :okexcept:
+
+    dt["child-node"].children = {"new-child": dt}
+
+Alternatively you can also create a :py:class:`~xarray.DataTree` object from:
+
+- A dictionary mapping directory-like paths to either :py:class:`~xarray.DataTree` nodes or data, using :py:meth:`xarray.DataTree.from_dict()`,
+- A well formed netCDF or Zarr file on disk with :py:func:`~xarray.open_datatree()`. See :ref:`reading and writing files <io>`.
+
+For data files with groups that do not not align see
+:py:func:`xarray.open_groups` or target each group individually
+:py:func:`xarray.open_dataset(group='groupname') <xarray.open_dataset>`. For
+more information about coordinate alignment see :ref:`datatree-inheritance`
+
+
+
+DataTree Contents
+~~~~~~~~~~~~~~~~~
+
+Like :py:class:`~xarray.Dataset`, :py:class:`~xarray.DataTree` implements the python mapping interface,
+but with values given by either :py:class:`~xarray.DataArray` objects or other
+:py:class:`~xarray.DataTree` objects.
+
+.. ipython:: python
+
+    dt["child-node"]
+    dt["foo"]
+
+Iterating over keys will iterate over both the names of variables and child nodes.
+
+We can also access all the data in a single node, and its inherited coordinates, through a dataset-like view
+
+.. ipython:: python
+
+    dt["child-node"].dataset
+
+This demonstrates the fact that the data in any one node is equivalent to the
+contents of a single :py:class:`~xarray.Dataset` object. The :py:attr:`DataTree.dataset <xarray.DataTree.dataset>` property
+returns an immutable view, but we can instead extract the node's data contents
+as a new and mutable :py:class:`~xarray.Dataset` object via
+:py:meth:`DataTree.to_dataset() <xarray.DataTree.to_dataset>`:
+
+.. ipython:: python
+
+    dt["child-node"].to_dataset()
+
+This same call can be made to get only the local node variables without any
+inherited ones, by setting the inherited keyword to False, but in this example
+there are no inherited coordinates so the result is the same as the previous call.
+
+.. ipython:: python
+
+    dt["child-node"].to_dataset(inherited=False)
```

Comment:
Consider omiting this here, and leaving the discussion of `inherited=False` to places where we can give a working example.